### PR TITLE
To the swift result

### PR DESCRIPTION
**Pull request by AndreaRR18**
Replaced our enum with the system one. 
Fix all the stencil with new Swift Result.

@ChrisMyrants @broomburgo Please review.

### CODE REVIEW

**Review by ChrisMyrants — APPROVED** [review]
@AndreaRR18 done, good job

File: Sources/FunctionalKit/Result.swift [hunk 1]
```diff
@@ -35,17 +34,15 @@ public enum Result<Failure,Parameter> where Failure: Error {
     }
 }
 
-extension Result: Equatable where Failure: Equatable, Parameter: Equatable {}
-
 extension Result: CoproductType {
     public typealias LeftType = Failure
-    public typealias RightType = Parameter
+    public typealias RightType = Success
```
Comment (broomburgo) [inline Sources/FunctionalKit/Result.swift]: `LeftType` and `RightType` should be inverted.

File: Sources/FunctionalKit/Result.swift [hunk 2]
```diff
@@ -67,34 +64,16 @@ extension Result: PureConstructible {
 }
 
 public extension Result {
-    typealias Generic<F,A> = Result<F,A> where F: Error
+    typealias Generic<F,A> = Result<A,F> where F: Error
```
Comment (broomburgo) [inline Sources/FunctionalKit/Result.swift]: `Generic` type parameters should have the same order as `Result` ones.

File: Sources/FunctionalKit/Result.swift [hunk 3]
```diff
@@ -5,18 +5,17 @@ import Abstract
...
 // sourcery: testMonad
 // sourcery: testConstruct = "success(x)"
 // sourcery: testSecondaryParameter
-public enum Result<Failure,Parameter> where Failure: Error {
-    case success(Parameter)
-    case failure(Failure)
+extension Result {
     
-    public func run() throws -> Parameter {
+    public func run() throws -> Success {
```
Comment (broomburgo) [inline Sources/FunctionalKit/Result.swift]: This is redundant now. Swift `Result` already includes a `get()` method that does the same thing.

**Review by broomburgo — CHANGES_REQUESTED** [review]

**Review by broomburgo — APPROVED** [review]

**Pull request merged**